controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -82,6 +82,14 @@ var (
 	// NOTE: The following assumes that kubeadm version equals to Kubernetes version.
 	minVerUnversionedKubeletConfig = semver.MustParse("1.24.0")
 
+	// MinKubernetesVersionImageRegistryMigration is first kubernetes version where
+	// images are only available at registry.k8s.io instead of k8s.gcr.io.
+	MinKubernetesVersionImageRegistryMigration = semver.MustParse("1.25.0")
+
+	// NextKubernetesVersionImageRegistryMigration is the next minor version after
+	// migration the image registry to registry.k8s.io.
+	NextKubernetesVersionImageRegistryMigration = semver.MustParse("1.26.0")
+
```

Comment:
Please unexport the constants.

> images are only available at registry.k8s.io instead of k8s.gcr.io.

That is not correct. Images are still available in both as one is just a redirect to the other. It's more like that's the version where v1.25.0 is the default.

It's just a special case in CAPD tests that we rely on the images being baked in with the correct registry, but I would not mention that here.

Reply:
:+1: adjusted the comment and also made the vars unexported :-) I did use them used directly in the controller but since adding the `ImageRepositoryFromClusterConfig` func it does not need to get exported anymore 👍